analyses/gistic-cohort-vs-histology-comparison/gistic-cohort-vs-histology-comparison.Rmd

```diff
@@ -0,0 +1,342 @@
+
+```{r fig.height = 10, fig.width = 24}
+lgat_plot_list <- list(lgat_scores_plot, cohort_scores_plot)
+generate_multipanel_plot(plot_list = lgat_plot_list,
+                         plot_title = "LGAT scores versus entire cohort",
+                         output_directory = plots_dir,
+                         output_filename = "lgat_gistic_scores_multipanel_plot.png")
+
+hgat_plot_list <- list(hgat_scores_plot, cohort_scores_plot)
+generate_multipanel_plot(plot_list = hgat_plot_list,
+                         plot_title = "HGAT scores versus entire cohort",
+                         output_directory = plots_dir,
+                         output_filename = "hgat_gistic_scores_multipanel_plot.png")
+
+medulloblastoma_plot_list <- list(medulloblastoma_scores_plot, cohort_scores_plot)
+generate_multipanel_plot(plot_list = medulloblastoma_plot_list,
+                         plot_title = "Medulloblastoma scores versus entire cohort",
+                         output_directory = plots_dir,
+                         output_filename = "medulloblastoma_gistic_scores_multipanel_plot.png")
+```
+
+## Compare `amp_genes.conf_90.txt` and `del_genes.conf_90.txt` files
+
+### Custom function
+```{r}
+plot_amp_del_genes <-
+  function(cohort_genes_file,
+           histology_genes_file) {
+    # Given the GISTIC result `amp_genes_conf_90.txt` or `del_genes_conf_90.txt`
+    # files for the entire cohort and a specific histology, plot a Venn Diagram
+    # showing the counts of rows that overlap/do not overlap between the two
+    # files.
+    #
+    # Args:
+    #   cohort_genes_file: the file path to the amp/del genes file for
+    #                           the entire cohort
+    #   histology_genes_file: the file path to the amp/del genes file
+    #                              for an individual histology
+    #
+    # Returns:
+    #  This function displays a Venn Diagram that represents the data that
+    #  overlaps/does not overlap between the two given files
+    
+    # Read in files
+    cohort_genes <-
+      data.table::fread(cohort_genes_file,
+                        data.table = FALSE)
+    
+    histology_genes <-
+      data.table::fread(histology_genes_file,
+                        data.table = FALSE)
+    
+    # Wrnagle data in cohort file for plotting
+    cohort_genes <- t(cohort_genes)
+    
+    cohort_genes <- cohort_genes[-1,] %>%
+      as.data.frame() %>%
+      dplyr::mutate(summary_info = paste0(V1, ", ", V2, ", ", V3, ", ", V4)) %>%
+      dplyr::select(-c("V1", "V2", "V3", "V4")) %>%
+      dplyr::select(summary_info, dplyr::everything()) %>%
+      tidyr::gather(V5, V6,-summary_info) %>%
+      dplyr::group_by(summary_info) %>%
+      dplyr::mutate(genes = paste0(sort(unique(V6)), collapse = ", ")) %>%
+      dplyr::select(-V5,-V6) %>%
+      dplyr::distinct() %>%
+      dplyr::mutate(summary = paste0(summary_info, genes)) %>%
+      dplyr::filter(!(summary == "NA, NA, NA, NA")) %>%
+      dplyr::arrange()
```

Comment:
For the amplification and deletion files, GISTIC produces "ragged columns" per [their documentation](https://www.genepattern.org/modules/docs/GISTIC_2.0). It is often helpful to use lists any time you find yourself in a situation where columns or rows are of different sizes and you have a whole bunch of trailing blanks.

Here's what I came up with for the first function mentioned above, getting the vector of genes of interest for the Venn diagram:

```r
get_genes_vector <- function(genes_file) {
  genes_ragged_df <- data.table::fread(genes_file,
                                       data.table = FALSE)
  genes_list <- as.list(genes_ragged_df)
  genes_list <- genes_list %>%
    # This removes any element from the list that is all NA -- most likely a
    # result of reading in a ragged array
    purrr::discard(~ all(is.na(.))) %>%
    # This removes the element of the list that is essentially the "header" for
    # the file
    purrr::discard(~ any(str_detect(., "cytoband|q value"))) %>% 
    # Remove any broad peaks with q-value > 0.05
    purrr::discard(~ .[3] > 0.05) %>%
    # Remove everything before and including the wide peak boundaries for each
    # remaining element of the list
    purrr::modify(~ .[-c(1:(str_which(., "chr")))]) %>%
    # Remove blanks -- result of ragged data.frame
    purrr::modify(~ .[. != ""])
  # This will give us the vector of all the genes that were included
  genes_vector <- unique(unname(unlist(genes_list)))
}
```

Reply:
Noted, I'll give this function a try and commit the changes.